xtask/src/cli.rs

```diff
@@ -0,0 +1,65 @@
+use clap::{Parser, Subcommand};
+
+#[derive(Parser, Debug)]
+#[command(version, about, long_about = None)]
+pub struct Cli {
+    #[command(subcommand)]
+    pub command: Commands,
+}
+
+#[derive(Subcommand, Debug)]
+pub enum Commands {
+    Test(test::Params),
+}
+
+pub mod test {
+    use clap::Args;
+    #[derive(Debug, Args)]
+    pub struct Params {
+        #[clap(flatten)]
+        pub build: BuildParams,
+
+        #[clap(flatten)]
+        pub shim: ShimParams,
+
+        #[clap(flatten)]
+        pub zombienet: ZombienetParams,
+
+        #[clap(flatten)]
+        pub sovereign: SovereignParams,
+    }
+
+    #[derive(clap::Args, Debug, Clone)]
+    pub struct BuildParams {
+        #[clap(default_value = "false")]
+        #[arg(long = "build-skip", value_name = "skip", id = "build.skip")]
+        pub skip: bool,
+
+        #[arg(long = "build-quiet", value_name = "quiet", id = "build.quiet")]
+        pub quiet: bool,
+    }
+
+    #[derive(clap::Args, Debug, Clone)]
+    pub struct ShimParams {
+        // TODO:
+        // This solution is a workaround for what I had initially thought.
+        // I would prefer to have something like `cargo xtask test --zombienet <zombienet_options>`
+        // instead of multiple options with just the prefix 'zombienet'.
+        // That is why in the future, we may add more flags,
+        // so having the structs already seems like a good idea
+        #[arg(long = "shim-quiet", value_name = "quiet", id = "shim.quiet")]
+        pub quiet: bool,
+    }
+
+    #[derive(clap::Args, Debug, Clone)]
+    pub struct ZombienetParams {
+        #[arg(long = "zombienet-quiet", value_name = "quiet", id = "zombienet.quiet")]
+        pub quiet: bool,
+    }
+
+    #[derive(clap::Args, Debug, Clone)]
+    pub struct SovereignParams {
+        #[arg(long = "sovereign-quiet", value_name = "quiet", id = "sovereign.quiet")]
```

Comment:
Why would you need id?

Reply:
without it clap panics at runtime:

```shell
Compiling xtask v0.1.0 (/home/gabriele/Documents/thrum/blobs/xtask)
    Finished dev [unoptimized] target(s) in 9.89s
     Running `target/debug/xtask test --build-skip`
thread 'main' panicked at /home/gabriele/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_builder-4.4.17/src/builder/debug_asserts.rs:86:13:
Command test: Argument names must be unique, but 'quiet' is in use by more than one argument or group
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

this is just a work around though, I didn't want to spend a lot of time fighting against clap::derive so I decided to go for the easiest solution for now

any other solution is extremely welcome

Reply:
Another solution could be to set a prefix for all the fields in test::Params (if there is a way to do that in clap).

Reply:
it doesn't seem like each `Params` struct should have its own `quiet` param and there should rather be a single `quiet` flag in the `BuildParams`

Reply:
For sure, this would simplify a lot of things, but I find this kind of granularity necessary. The reason I added them is that now all the stdout is printed in the same terminal. This means zombienet, shim, and sovereign things are all displayed one on top of the other, and this makes debugging extremely cumbersome (i.e., manually redirecting stdout and grepping things). Having the ability to filter out which program to print simplifies things.

Another approach could be, instead of quiet, adding a flag to decide whether or not to redirect stdout to files, like zombienet does automatically with `zombienet/collator01.log` (or keep both approaches)

Reply:
I, tbh, feel that the appropriate solution here would be to act as zombienet (and it's predecessor polkadot-launch) and redirect the output of the children to some predefined files. In stdout you would just write a heads up where the user can find the logs.

Reply:
Ok, I like it! I will implement it right now!

Reply:
@rphmeier, if it's okay with you, for me you can merge it now and I will address the redirect into log files in another PR. I tried implementing it today, but I encountered some strange behavior

Reply:
Sure, that seems fine. You can already start building on top of this branch using Graphite (actually, that's the point)